docs/code/SchemaComprehension.md

```diff
@@ -0,0 +1,210 @@
+# Schema comprehension in ML.NET
+
+This document describes in detail the under-the-hood mechanism that ML.NET uses to automate the creation of `IDataView` schema, with the goal to make it as convenient to the end user as possible, while not incurring extra computational costs.
+
+## Introduction
+
+Every dataset in ML.NET is an `IDataView`, which is, for the purposes of this document, a collection of rows that share the same columns. The set of columns, their names, types and other metadata is known as the *schema* of the `IDataView`, and it's represented as an `ISchema` object.
+
+Before any new data enters ML.NET, the user needs to somehow define how the schema of the data will look like.
+To do this, the following questions need to be answered:
+- What are the column names?
+- What are their types?
+- What other metadata is associated with the columns?
+
+These items above are very similar to the definition of fields in a C# class: names and types of columns correspond to names and types of fields, and metadata can correspond to field attributes. 
+Because of this similarity, ML.NET offers a common convenient mechanism for creating a schema: it is done via defining a C# class.
+
+For example, the below class definition can be used to define a data view with 5 float columns:
```

Comment:
>data view [](start = 64, length = 9)

wondering if it would help to state at the beginning that IDataView and 'data view' are interchangeable, because you give the definition of one, and use the other term for it.